odoo/tools/date_utils.py

```diff
@@ -1,39 +1,43 @@
 # -*- coding: utf-8 -*-
-import math
 import calendar
+import math
 from datetime import date, datetime, time
+from typing import TypeVar
+
 import pytz
 from dateutil.relativedelta import relativedelta
 
 from . import ustr
 from .func import lazy
 
+D = TypeVar('D', date, datetime)
+
 def get_month(date):
```

Comment:
Well here I "merely" wanted to fix the nonsense types because they were bothering me and triggering pycharm warnings. Also I've never really looked into the type-annotation PEPs so I don't really know the details or the cross-version issues (e.g. I've seen that sometimes the annotations are passed as strings and some versions delay evaluation but others not).

docstrings I understand :D